### PR TITLE
feat: Add data integrity check to identify duplicate category option combos [DHIS2-18258]

### PR DESCRIPTION
**Pull request by david-mackessy**
Adds a data integrity check to check for duplicate category option combos.
A duplicate category option combo has the same category options as another category option combo, which are both members of the same category combo.

- check added
- tests added
- tested manually as well and seems to work as expected
- i18n mapping added

@jason-p-pickering provided the SQL 👍

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-administration/src/main/resources/data-integrity-checks/categories/category_option_combos_same_category_options.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      INNER JOIN (
+      SELECT catoptions, COUNT(*) as count, DENSE_RANK() OVER (ORDER BY catoptions) as rank
+      FROM ( SELECT categoryoptioncomboid,
+      array_agg(categoryoptionid ORDER BY categoryoptionid) as catoptions
+      FROM categoryoptioncombos_categoryoptions group by categoryoptioncomboid) as foo
+      GROUP BY catoptions HAVING COUNT(*) > 1
+      ) b on b.catoptions = a.catoptions
+      ORDER BY b.rank;
+  severity: WARNING
+  introduction: >
```
Comment (david-mackessy) [inline dhis-2/dhis-services/dhis-service-administration/src/main/resources/data-integrity-checks/categories/category_option_combos_same_category_options.yaml:59]: @jason-p-pickering does this intro make sense?

**Comment by david-mackessy** [comment]
@jason-p-pickering If you could review this new check that'd be great!

File: dhis-2/dhis-services/dhis-service-administration/src/main/resources/data-integrity-checks/categories/category_option_combos_same_category_options.yaml (hunk 1 above)
Comment (jason-p-pickering) [reply dhis-2/dhis-services/dhis-service-administration/src/main/resources/data-integrity-checks/categories/category_option_combos_same_category_options.yaml]: Redid the SQL a bit so OBE

File: dhis-2/dhis-services/dhis-service-administration/src/main/resources/data-integrity-checks/categories/category_option_combos_same_category_options.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    Category option combos with the exact same category options should be considered
+    to be merged. Category option combos with the exact same category options
+    may be easily confused by users in analysis. The details view will
+    provide a list of category option combos which have the exact same category options.
+    For each duplicated category option combo a number in parentheses such as (1) will
+    indicate which category option combos belong to duplicated groups.
+  details_id_type: categoryOptionCombos
+  recommendation: |
```
[the remark this answers is not in the document]
Comment (jason-p-pickering) [reply dhis-2/dhis-services/dhis-service-administration/src/main/resources/data-integrity-checks/categories/category_option_combos_same_category_options.yaml]: Added in my PR into this branch.

**Review by jbee — APPROVED** [review]

**Comment by jason-p-pickering** [comment]
Looks good to me now but asking for an external review from  @stian-sandvold since I wrote part of this code.

**Review by vietnguyen — APPROVED** [review]

**Pull request merged**